src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs

```diff
@@ -157,55 +158,55 @@ public override string ToString()
 
         public override bool Equals(object? obj)
         {
-            if (!IsGenericMethod)
-                return obj == (object)this;
+            if (IsGenericMethod)
+            {
+                // We cannot do simple object identity comparisons for generic methods.
+                // Equals will be called in CerHashTable when RuntimeType+RuntimeTypeCache.GetGenericMethodInfo()
+                // retrieve items from and insert items into s_methodInstantiations which is a CerHashtable.
 
-            // We cannot do simple object identity comparisons for generic methods.
-            // Equals will be called in CerHashTable when RuntimeType+RuntimeTypeCache.GetGenericMethodInfo()
-            // retrieve items from and insert items into s_methodInstantiations which is a CerHashtable.
+                RuntimeMethodInfo? mi = obj as RuntimeMethodInfo;
 
-            RuntimeMethodInfo? mi = obj as RuntimeMethodInfo;
+                if (mi == null || !mi.IsGenericMethod)
+                    return false;
 
-            if (mi == null || !mi.IsGenericMethod)
-                return false;
+                // now we know that both operands are generic methods
 
-            // now we know that both operands are generic methods
+                IRuntimeMethodInfo handle1 = RuntimeMethodHandle.StripMethodInstantiation(this);
+                IRuntimeMethodInfo handle2 = RuntimeMethodHandle.StripMethodInstantiation(mi);
+                if (handle1.Value.Value != handle2.Value.Value)
+                    return false;
 
-            IRuntimeMethodInfo handle1 = RuntimeMethodHandle.StripMethodInstantiation(this);
-            IRuntimeMethodInfo handle2 = RuntimeMethodHandle.StripMethodInstantiation(mi);
-            if (handle1.Value.Value != handle2.Value.Value)
-                return false;
+                Type[] lhs = GetGenericArguments();
+                Type[] rhs = mi.GetGenericArguments();
 
-            Type[] lhs = GetGenericArguments();
-            Type[] rhs = mi.GetGenericArguments();
+                if (lhs.Length != rhs.Length)
+                    return false;
 
-            if (lhs.Length != rhs.Length)
-                return false;
+                for (int i = 0; i < lhs.Length; i++)
+                {
+                    if (lhs[i] != rhs[i])
+                        return false;
+                }
 
-            for (int i = 0; i < lhs.Length; i++)
-            {
-                if (lhs[i] != rhs[i])
+                if (DeclaringType != mi.DeclaringType)
                     return false;
-            }
 
-            if (DeclaringType != mi.DeclaringType)
-                return false;
+                if (ReflectedType != mi.ReflectedType)
+                    return false;
 
-            if (ReflectedType != mi.ReflectedType)
-                return false;
+                return true;
+            }
+
+            return obj == (object)this ||
```

Comment:
Probably as change to `ReferenceEquals` as https://github.com/dotnet/runtime/pull/73009/files#r935802561